jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenSettingsProxyProvider.java

```diff
@@ -18,33 +18,62 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.List;
 import javax.annotation.Nullable;
+import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.settings.Proxy;
+import org.apache.maven.settings.Settings;
+import org.apache.maven.settings.building.SettingsProblem;
+import org.apache.maven.settings.crypto.DefaultSettingsDecryptionRequest;
+import org.apache.maven.settings.crypto.SettingsDecrypter;
+import org.apache.maven.settings.crypto.SettingsDecryptionRequest;
+import org.apache.maven.settings.crypto.SettingsDecryptionResult;
 
 /** Propagates proxy configuration from Maven settings to system properties. */
-class ProxyProvider {
+class MavenSettingsProxyProvider {
 
   private static final ImmutableList<String> PROXY_PROPERTIES =
       ImmutableList.of("proxyHost", "proxyPort", "proxyUser", "proxyPassword");
 
   /**
-   * Initializes proxy settings based on Maven settings.
+   * Initializes proxy settings based on Maven settings if they are not already set by the user
+   * directly.
    *
    * @param settings Maven settings
    */
-  static void init(DecryptedMavenSettings settings) {
-    configureProxy(settings, "https");
-    configureProxy(settings, "http");
-  }
+  static void activateHttpAndHttpsProxies(Settings settings, SettingsDecrypter decrypter)
+      throws MojoExecutionException {
+    List<Proxy> proxies = new ArrayList<>(2);
+    for (String protocol : ImmutableList.of("http", "https")) {
+      if (areProxyPropertiesSet(protocol)) {
+        return;
```

Comment:
`continue;`

Reply:
ugh, why are the tests passing? I'm so confused.

Reply:
I'm not confused, I need to write a few more tests.

Reply:
done

Reply:
:man_facepalming: